CONTENT-5678 - Mongoose vs MongoDB Driver

kukicado opened the pull request:
Preview URL:  http://docs-devhub-staging.s3-website-us-east-1.amazonaws.com/908a194/devhub/docsworker-xlarge/mongoose-node-js-driver/article/mongoose-versus-node-js-driver

kenwalger commented on source/article/mongoose-versus-node-js-driver.txt:
```diff
@@ -0,0 +1,330 @@
+   
+   `Mongoose <https://mongoosejs.com/>`__ is a Node.js based Object Data Modeling (ODM) library for MongoDB. It is akin to an Object Relational Mapper (ORM) such as `SQLAlchemy <https://www.sqlalchemy.org/>`__ for traditional SQL databases. The problem that Mongoose aims to solve is allowing developers to enforce a specific schema at the application layer. In addition to enforcing a schema, Mongo ...
+
+   In the past, the only way to enforce a schema against a MongoDB collection was to do it at the application level using an ODM like Mongoose, but as of MongoDB 3.2, there is perhaps a better way to do schema validation at the database. In this article, we'll take a look at `MongoDB Schema Validation <https://docs.mongodb.com/manual/core/schema-validation/>`__ and how it can help us enforce a da ...
```
_at the database._ Maybe _at the database *level*_?

kenwalger commented on source/article/mongoose-versus-node-js-driver.txt:
```diff
@@ -0,0 +1,330 @@
+          console.log(post);
+      })
+
+   The benefit of this approach is that we have a schema to work against in our application code and an explicit relationship between our MongoDB documents and the  Mongoose models within our application. The downside, we can only create blog posts and they have to follow the above defined schema. If we change our Mongoose schema, we are changing the relationship completely, and if you're going t ...
```
Remove extra space between **the** and **Mongoose models within our application.**

MaBeuLux88 commented on source/article/mongoose-versus-node-js-driver.txt:
```diff
@@ -0,0 +1,332 @@
+.. twitter::
+   :site: @mongodb
+   :creator: @kukicado
+   :title: Do You Need Mongoose When Developing Node.js and MongoDB Applications?
```
Missing twitter image & image-alt
https://wiki.corp.mongodb.com/display/DE/devhub-article+Reference

MaBeuLux88 replied:
Also missing the new "og" directive.

kukicado replied:
Addressed.

MaBeuLux88 commented:
http://docs-devhub-staging.s3-website-us-east-1.amazonaws.com/c4f4fd7/devhub/docsworker-xlarge/mongoose-node-js-driver/

I updated this branch with a `git fetch upstream && git rebase upstream/master` @kukicado. Care when you pull this branch, it will probably conflict with your local version.
I suggest you grab everything from the server with `git fetch && git reset --hard mongoose-node-js-driver` if you don't have anything to lose on this branch (that wasn't pushed already).
It allowed me to run the newest version of the `sanity-check.py`.

- The twitter title is 1 character too long.
- missing image and image-alt in twitter directive.
- missing the og directive and all the details in it. 

Apart from this, everything else looks fine from the script point of view.

kukicado commented:
Addressed the feedback and added all the metadata. We should be good to merge tomorrow.

Preview URL: http://docs-devhub-staging.s3-website-us-east-1.amazonaws.com/9eb3068/devhub/docsworker-xlarge/mongoose-node-js-driver/article/mongoose-versus-node-js-driver
